### PR TITLE
Fix some memory leaks (FlxTransitionableState was a HUGE one)

### PR DESCRIPTION
**Pull request by larsiusprime**

### CODE REVIEW

File: flixel/addons/display/shapes/FlxShapeArrow.hx [hunk 1]
```diff
@@ -77,11 +77,15 @@ class FlxShapeArrow extends FlxShape
...
 	}
 	
-	override public function destroy():Void
+	override public function destroy():Void 
 	{
+		FlxDestroyUtil.destroy(point);
+		FlxDestroyUtil.destroy(point2);
+		FlxDestroyUtil.destroyArray(_vertices);
+		_vertices = null;
+		_matrix2 = _matrix;
```
Comment (gamedevsam) [inline flixel/addons/display/shapes/FlxShapeArrow.hx]: I don't understand this here, why assign matrix to eachother in destroy?

Comment (larsiusprime) [inline flixel/addons/display/shapes/FlxShapeArrow.hx]: supposed to be null.

**Comment by larsiusprime** [comment]
Fixed, got my wires crossed.

**Comment by Gama11** [comment]
So the destroy functions actually have a measurable impact? There was always some debate whether they're still necessary with modern GC.

**Comment by larsiusprime** [comment]
I don't know if the destroy functions have any impact or not... honestly the main issues in this function are the transition bits, skipping the super and leaving the callback intact where causing entire states to persist.

**Comment by Gama11** [comment]
I see, makes sense.

**Comment by MSGhero** [comment]
Lol, I meant the syntax to destroy is 'thing = FlxDestroyUtil.destroy(thing)', although I usually just destroy things myself and set them to null after.

**Pull request merged**